kwave/data.py

```diff
@@ -1,10 +1,55 @@
 from dataclasses import dataclass
+from typing import Any
 
 import numpy as np
 
 
+class Vector(np.ndarray):
+    def __new__(cls, elements: list):
+        elements = list(elements)[:3]  # Truncate the input list to have at most 3 elements
```

Comment:
would it be better to assert input list has 3 elements?

Reply:
Added the assertion and removed logic to take the first three elements